File: 剑指31.java
class Solution {
    public boolean validateStackSequences(int[] pushed, int[] popped) {
        Stack<Integer> stack = new Stack<>();
        int idx = 0;
        for(int i = 0; i < pushed.length; i++) {
            stack.push(pushed[i]);
            while(!stack.empty() && stack.peek() == popped[idx]){
                stack.pop();
                idx++;
            }
        }
        return stack.empty();
    }
}

File: 剑指31.py
class Solution:
    def validateStackSequences(self, pushed: List[int], popped: List[int]) -> bool:
        stack, idx = [], 0
        for num in pushed:
            stack.append(num)
            while stack and stack[-1]== popped[idx]:
                stack.pop()
                idx+=1
        return not stack
